pygfx/renderers/wgpu/engine/blender.py

```diff
@@ -911,3 +911,74 @@ def _create_combination_bind_group(self, bind_group_layout):
         return self.device.create_bind_group(
             layout=bind_group_layout, entries=bind_group_entries
         )
+
+
+class AdditivePass(BasePass):
+    """A pass that renders fragments with additive blending."""
+
+    render_mask = 3  # opaque end transparent
+    write_pick = False
+
+    def get_color_descriptors(self, blender, material_write_pick):
+        bf, bo = wgpu.BlendFactor, wgpu.BlendOperation
+        return [
+            {
+                "format": blender.color_format,
+                "blend": {
+                    "alpha": (bf.one, bf.one, bo.add),
+                    "color": (bf.one, bf.one, bo.add),
+                },
+                "write_mask": wgpu.ColorWrite.ALL,
+            },
+        ]
+
+    def get_color_attachments(self, blender):
+        color_load_op = wgpu.LoadOp.load
+        if blender.color_clear:
+            blender.color_clear = False
+            color_load_op = wgpu.LoadOp.clear
+
+        return [
+            {
+                "view": blender.color_view,
+                "resolve_target": None,
+                "load_op": color_load_op,
+                "store_op": wgpu.StoreOp.store,
+            },
+        ]
+
+    def get_depth_descriptor(self, blender):
+        return {
+            "format": blender.depth_format,
+            "depth_write_enabled": False,
+            "depth_compare": wgpu.CompareFunction.always,
+        }
+
+    def get_depth_attachment(self, blender):
+        depth_load_op = wgpu.LoadOp.load
+        if blender.depth_clear:
+            blender.depth_clear = False
+            depth_load_op = wgpu.LoadOp.clear
+        return {
+            "view": blender.depth_view,
+            "depth_load_op": depth_load_op,
+            "depth_store_op": wgpu.StoreOp.store,
+        }
+
+    def get_shader_code(self, blender):
+        return """
+        struct FragmentOutput {
+            @location(0) color: vec4<f32>,
+        };
+        fn get_fragment_output(depth: f32, color: vec4<f32>) -> FragmentOutput {
+            var out : FragmentOutput;
+            out.color = color;
```

Comment:
i did have one question about alpha, here and in the line `"alpha": (bf.one, bf.one, bo.add),`.  I'm aware that there are varying strategies for adding alpha (pre-multiplied, etc).  Is the current implementation what you would expect for a "standard" additive blending mode?  should i handle alpha differently, and/or be more specific in the naming of the blending mode?

Reply:
### Context

In e.g. the transparent pass, the color is premultiplied with the alpha:
```py
out.color = vec4<f32>(color.rgb * color.a, color.a);
```
With the blend function ...
```py
"blend": {
                    "alpha": (bf.one, bf.one_minus_src_alpha, bo.add),
                    "color": (bf.one, bf.one_minus_src_alpha, bo.add),
                },
```
... this results in a color that is `src_a * src_rgb + (1-src_a) * target_rgb`. I can't remember exactly why I did it this way, ad not use `(bf.src_alpha, bf.one_minus_src_alpha, bo.add)`, but one reason could be thatit results in the blend-functions for alpha and color to be equal.

### Back to your question

I haven't done anything with additive blending so I'm not sure what the expected behavior is. But I *think* the idea is that it all just adds up, so the current approach is correct.